docs/proposal/content-library-support.md

```diff
@@ -0,0 +1,257 @@
+* If both host and item are present, then host is the content library name (or uuid) and path is the item name
+* If more than one item is returned, this should be an error
+* If more than one content library with the same name is returned, it should be an error and users should use the Content Library ID
+
+```go
+import (
+    ...
+   	"github.com/vmware/govmomi/vapi/library"
+    ...
+)
+
+func getContentLibraryItem(someargs) (itemID string, err error) {
+    var clibrary, citem string
+    u, err := url.Parse(itemToSearch)
+    if err != nil {
+    	log.Fatal(err)
+    }
+    
+    citem = u.Host
+    if u.Path != "" {
+    	clibrary = u.Host
+    	citem = strings.TrimPrefix(u.Path, "/") // Remove leading prefix /
+    }
+    
+    m := library.NewManager(restclient)
+    
+    if clibrary != "" {
+        // Try to get Content Library by ID
+    	_, err = m.GetLibraryByID(ctx, clibrary)
+    	if err != nil {
+    		// In this case, we should get all libraries with a specific name,
+    		libs, err := m.FindLibrary(ctx, library.Find{
+    			Name: clibrary,
+    		})
+    		if err != nil {
+                // ERROR on getting library			
+    		}
+    		if len(libs) != 1 {
+    			// ERROR as no library or too many libraries has been found
+    		}
+    		clibrary = libs[0]
+    	}
+    }
+    
+    items, err := m.FindLibraryItems(ctx, library.FindItem{
+    	Name:      citem,
+    	Type:      "ovf",
+    	LibraryID: clibrary,
+    })
+    if err != nil {
+        // ERROR getting the Library items	
+    }
+    
+    if len(items) != 1 {
+        // ERROR no items or too many items with same name. In this case, should be more specific and pass
+        // a content library name
+    }
+    
+    return items[0], nil
+}
+```
+
+##### VSphereVM clone changes
+
+We need to enhance the [clone](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/19c5deb79e97a8cd9d546ec39db3f29231bbcb91/pkg/services/govmomi/vcenter/clone.go#L75) function to recogize the scheme we defined, and choose appropriate API to clone the VM.
+
+Once defined if Content Library should be used, a new specific clone function should be used,
+to make the proper cloning, but using vCenter Manager client instead of template/VM manager.
+
+The "customization" function can and should be reused, and decoupled from the main clone process. Instead 
+after "customizing" it should be passed to the proper cloning function (library item or template)
+
+```go
+// Based on https://github.com/vmware-tanzu/vm-operator/blob/main/pkg/vmprovider/providers/vsphere/session/session_vm_create.go and
+// https://github.com/vmware/govmomi/blob/main/govc/library/deploy.go
+import (
+    ...
+   	"github.com/vmware/govmomi/vapi/vcenter"
+    ...
+)
+
+func deployContentLibraryAsVM(itemID string, additionalargs...) (*object.VirtualMachine, error) {
+	m := vcenter.NewManager(restclient)
+	ref, err := m.DeployLibraryItem(ctx, itemID string, vcenter.Deploy{
```

Comment:
This function call is basically the equivalent to calling `tpl.Clone` right? (https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/19c5deb79e97a8cd9d546ec39db3f29231bbcb91/pkg/services/govmomi/vcenter/clone.go#L75)

Reply:
correct :) And the customization will need to happen differently, as we know what exists on a template, but the OVF deployment is different (I need to figure out actually if there's a way to get the spec and customize accordingly, I didn't wanted to go that down on the spec right now).

So, for template we call lastly `tpl.Clone` that generates a new task.

And for library we call DeployLibraryItem that does that directly (I'm checking if there's a way to generate a task as well)